x-pack/legacy/plugins/apm/index.ts

```diff
@@ -44,7 +44,7 @@ export const apm: LegacyPluginInitializer = kibana => {
           // TODO: rename to apm_oss.indexPatternTitle in 7.0 (breaking change)
           apmIndexPatternTitle: config.get('apm_oss.indexPattern'),
           apmServiceMapEnabled: config.get('xpack.apm.serviceMapEnabled'),
-          apmTransactionIndices: config.get('apm_oss.transactionIndices')
+          apmTransactionIndices: config.get('apm_oss.transactionIndices') // TODO address this!
```

Comment:
Is the problem that `injectDefaultVars` has to return synchronously?
Or could you fetch the saved object here?

Reply:
I do believe it has to be synchronous, but regardless: the issue here is that the value of `apm_oss.transactionIndices` could have changed via the UI Indices settings UI since it was injected here during plugin startup. So I'll have to make an API call for it whenever the client requires this index.

Reply:
Ahh yes ofc. Instead of making a separate endpoint for this, can you call the existing endpoint that exposes the ui indices?